File: 0957-minimum-add-to-make-parentheses-valid/solution.java
class Solution {
    public int minAddToMakeValid(String s) {
        int open=0,add=0;
        for(var ch:s.toCharArray()){
            if(ch=='('){
                open++;
            }else{
                if(open>0){
                    open--;
                }else if(open==0){
                    add++;
                }
            }
        }return open+add ;
    }
}

File: 0957-minimum-add-to-make-parentheses-valid/solution.cpp
class Solution {
public:
    int minAddToMakeValid(string s) {
        int open_paren_needed=0;
        int stack_size=0;
        for(char &ch:s){
            if(ch=='('){
                stack_size++;
            }else if(stack_size>0){
                stack_size--;
            }else{
                open_paren_needed++;
            }
        }return open_paren_needed+stack_size;

    }
};
